tests/saitests/py3/sai_qos_tests.py

```diff
@@ -1024,18 +1025,17 @@ def runTest(self):
                 print(list(map(operator.sub, pg_cntrs, pg_cntrs_base)),
                       file=sys.stderr)
                 for i in range(0, PG_NUM):
+                    # In DNX, ip2me packets trapped to cpu mapped to TC 1 -> PG 0 and
+                    # all other control packets trapped to cpu mapped to TC 4 -> PG 4.
                     if i == pg:
-                        if i == 0 or i == 4:
-                            assert (pg_cntrs[pg] >=
-                                    pg_cntrs_base[pg] + len(dscps))
+                        if platform_asic and platform_asic == "broadcom-dnx" and i in [0, 4, 7]:
+                            assert (pg_cntrs[pg] >= pg_cntrs_base[pg] + len(dscps))
                         else:
                             assert (pg_cntrs[pg] ==
                                     pg_cntrs_base[pg] + len(dscps))
                     else:
-                        # LACP packets are mapped to queue0 and tcp syn packets for BGP to queue4
-                        # So for those queues the count could be more
-                        if i == 0 or i == 4:
-                            assert (pg_cntrs[i] >= pg_cntrs_base[i])
+                        if platform_asic and platform_asic == "broadcom-dnx" and i in [0, 4, 7]:
```

Comment:
The behavior where we see packets in PG 7 of the source port, due to TX packets from CPU sent out with TC =7  --- **is Broadcom specific**

Shouldn't the check be something like this -- i.e expect PG7 only if it is brcm-dnx ? 
if i == 0 or i == 4 or (platform_asic and platform_asic == "broadcom-dnx" and i == 7):